### ISSUE TITLE
[java] MethodReturnsInternalArray doesn't consider lambda

### ISSUE THREAD

**Issue by Han-Cui** [pmd/pmd#4246]
**Affects PMD Version:** 6.52.0

**Rule:**[MethodReturnsInternalArray](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#methodreturnsinternalarray)

**Description:** Modified by the case given in [#3630](https://github.com/pmd/pmd/issues/3630#issue-1053968769). The anonymous class can be changed to lambda expression which return internal array. Maybe it is a false negative here.

**Code Sample demonstrating the issue:**

```java
public class Test {
    private static final String[] names = new String[] { "a", "b" };

    public static Provider getProvider() {
        return () -> names; // expected a violation here
    }

    public interface Provider {
        String[] getNames();
    }
}

```

**Expected outcome:**

PMD should report a violation at line 5, but doesn't. This is a false-negative.

**Running PMD through:** *CLI*
